### ISSUE TITLE
.map using a pipeline

### ISSUE THREAD

**Issue by bradserbu** [caolan/highland#531]
How would you preform a .map using a pipeline as the target?

**Comment by vqvu** [comment]
`pipeline` is useful for creating node-compatible through streams for use with `pipe`. If you're just working with Highland streams, you can skip the `pipeline` entirely.

```javascript
.map(record => $([record]).through(processArray))
```

**Comment by bradserbu** [comment]
This is great.  Thanks Victor.

**Comment by bradserbu** [comment]
@vqvu Is there anyway to accomplish this without creating the extra stream: `$([record]).through`? 

For instance, if the `.map` method were to support passing the `processArray` or a pipeline object it would look like:
```
return $(input).map(processArray).mergeWithLimit(concurrency)
```

Thanks.

**Comment by vqvu** [comment]
No. You need to create a stream in order to represent an asynchronous processing step. If `processArray` can operate directly on a `record` (rather than operating on a stream *containing* `record`), then you can an extra allocation and pass `processArray` directly to `map`. `processArray` would still need to return a stream, however.

**Comment by bradserbu** [comment]
Okay.  Thanks.
